### PR TITLE
DeepL: Default to tag_handling: xml

### LINKED ISSUE #417: DeepL translates variable names
When using the DeepL API via i18n-tasks, variable names sometimes end up translated as well. From looking at the code, there seems to be an attempt to avoid this, but it doesn't work reliably. As a rough estimate, ca. 10% of variables end up translated. It also seems to be the same variables regardless of output language, as I've tried this with English to both French and German now, with the same result.

### PR DESCRIPTION
**Pull request by davidwessman**
- Also asks DeepL to ignore the <i18n> tag which is used to wrap
  interpolated variables.
- Fixes #417

### CODE REVIEW

File: spec/deepl_translate_spec.rb [hunk 1]
```diff
@@ -6,15 +6,16 @@
 
 RSpec.describe 'DeepL Translation' do
   nil_value_test = ['nil-value-key', nil, nil]
-  text_test      = ['key', "Hello, %{user} O'Neill! How are you?", "¡Hola, %{user} O'Neill! ¿Cómo estás?"]
+  text_test      = ['key', "Hello, %{user} O'Neill! How are you?", "¡Hola, %{user} O'Neill! ¿Qué tal estás?"]
```
Comment (davidwessman) [inline spec/deepl_translate_spec.rb:9]: Not sure why this changes :/ and Spanish is not my forté, I googled it and it seems like it might be right.

Comment (glebm) [reply spec/deepl_translate_spec.rb:9]: Well that's definitely not right but that's OK it's just a test.

File: lib/i18n/tasks/translators/deepl_translator.rb [hunk 2]
```diff
@@ -41,11 +41,11 @@ def options_for_translate_values(**options)
     end
 
     def options_for_html
-      { tag_handling: 'xml' }
+      { tag_handling: 'xml', ignore_tags: %w[i18n] }
     end
 
     def options_for_plain
-      { preserve_formatting: true }
+      { preserve_formatting: true, tag_handling: 'xml', ignore_tags: %w[i18n] }
```
Comment (glebm) [inline lib/i18n/tasks/translators/deepl_translator.rb:48]: Do we now need to html-escape/unescape plain text?

Comment (davidwessman) [reply lib/i18n/tasks/translators/deepl_translator.rb]: I added a suggestion on escaping + unescaping now 🙂

File: spec/deepl_translate_spec.rb (hunk 1 above)
Comment (davidwessman) [reply spec/deepl_translate_spec.rb:9]: Yeah, I guess it is the users problem when using interpolation :/

**Pull request merged**